macOS/DuckDuckGo/VisualRefresh/VisualStyleConfigurable.swift

```diff
@@ -21,18 +21,88 @@ import BrowserServicesKit
 import FeatureFlags
 
 protocol VisualStyleConfigurable {
-    var toolbarHeight: CGFloat { get }
+    func addressBarHeight(for type: AddressBarSizeClass) -> CGFloat
+    func addressBarTopPaddig(for type: AddressBarSizeClass) -> CGFloat
+    func addressBarBottomPaddig(for type: AddressBarSizeClass) -> CGFloat
+}
+
+enum AddressBarSizeClass {
+    case `default`
+    case homePage
+    case popUpWindow
+
+    var logoWidth: CGFloat {
+        switch self {
+        case .homePage: 44
+        case .popUpWindow, .default: 0
+        }
+    }
+
+    var isLogoVisible: Bool {
+        switch self {
+        case .homePage: true
+        case .popUpWindow, .default: false
+        }
+    }
 }
 
 struct VisualStyle {
-    let toolbarHeight: CGFloat
+    private let addressBarHeightForDefault: CGFloat
+    private let addressBarHeightForHomePage: CGFloat
+    private let addressBarHeightForPopUpWindow: CGFloat
+    private let addressBarTopPaddingForDefault: CGFloat
+    private let addressBarTopPaddingForHomePage: CGFloat
+    private let addressBarTopPaddingForPopUpWindow: CGFloat
+    private let addressBarBottomPaddingForDefault: CGFloat
+    private let addressBarBottomPaddingForHomePage: CGFloat
+    private let addressBarBottomPaddingForPopUpWindow: CGFloat
+
+    func addressBarHeight(for type: AddressBarSizeClass) -> CGFloat {
+        switch type {
+        case .default: return addressBarHeightForDefault
+        case .homePage: return addressBarHeightForHomePage
+        case .popUpWindow: return addressBarHeightForPopUpWindow
+        }
+    }
+
+    func addressBarTopPaddig(for type: AddressBarSizeClass) -> CGFloat {
+        switch type {
+        case .default: return addressBarTopPaddingForDefault
+        case .homePage: return addressBarTopPaddingForHomePage
+        case .popUpWindow: return addressBarTopPaddingForPopUpWindow
+        }
+    }
+
+    func addressBarBottomPaddig(for type: AddressBarSizeClass) -> CGFloat {
+        switch type {
+        case .default: return addressBarBottomPaddingForDefault
+        case .homePage: return addressBarBottomPaddingForHomePage
+        case .popUpWindow: return addressBarBottomPaddingForPopUpWindow
+        }
+    }
 
     static var old: VisualStyle {
```

Comment:
Given that “new” will be the new “current” :D I would consider renaming them as “legacy” and “current”, and then renaming “currentStyle” as “activeStyle”, but I’m not strong on that.